portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/search/elasticsearch/Elasticsearch6EE.testcase

```diff
@@ -28,6 +28,171 @@ definition {
 		}
 	}
 
+	test AssertSimilarResultsMatchesAssetType {
+		property test.name.skip.portal.instance = "Elasticsearch7EE#AssertSimilarResultsMatchesAssetType";
+
+		for (var count : list "1,2,3") {
+			JSONWebcontent.addWebContent(
+				content = "WC Content",
+				groupName = "Guest",
+				title = "Similar Results Blogs Title ${count}");
+
+			JSONBlog.addEntry(
+				entryContent = "Blogs Content",
+				entryTitle = "Similar Results DM Title ${count}");
+
+			JSONMBMessage.addMessage(
+				groupName = "Guest",
+				threadBody = "MB Content",
+				threadSubject = "Similar Results MB Title ${count}");
+
+			JSONDocument.addFile(
+				dmDocumentDescription = "DM Content",
+				dmDocumentTitle = "Similar Results WC Title ${count}",
+				groupName = "Guest");
+		}
+
+		JSONLayout.addPublicLayout(
+			groupName = "Guest",
+			layoutName = "Similar Results Page");
+
+		JSONLayout.addWidgetToPublicLayout(
+			column = "1",
+			groupName = "Guest",
+			layoutName = "Similar Results Page",
+			widgetName = "Asset Publisher");
+
+		Navigator.gotoPage(pageName = "Similar Results Page");
+
+		Portlet.addPG(portletName = "Similar Results");
+
+		Portlet.gotoPortletOptions(
+			portletName = "Similar Results",
+			portletOption = "Configuration");
+
+		SelectFrame(locator1 = "IFrame#CONFIGURATION");
+
+		Select(
+			locator1 = "Select#DISPLAY_TEMPLATE",
+			value1 = "Compact Layout");
+
+		Type(
+			key_fieldLabel = "Minimum Document Frequency",
+			locator1 = "TextInput#GENERIC_TEXT_INPUT",
+			value1 = "1");
+
+		Type(
+			key_fieldLabel = "Minimum Term Frequency",
+			locator1 = "TextInput#GENERIC_TEXT_INPUT",
+			value1 = "1");
+
+		PortletEntry.save();
+
+		IFrame.closeFrame();
```

Comment:
Can we turn this into a macro? We duplicate these steps in the other test as well.